frontend/web/components/release-pipelines/CreateReleasePipeline.tsx

```diff
@@ -116,24 +109,36 @@ function CreateReleasePipeline() {
   ])
 
   useEffect(() => {
-    console.log('existingPipeline', existingPipeline)
     if (existingPipeline) {
       setPipelineData(existingPipeline)
     }
   }, [existingPipeline])
 
-  const handleOnChange = (newStageData: DraftStageType, index: number) => {
+  const handleOnChange = (
+    newStageData: PipelineStageRequest,
+    index: number,
+  ) => {
     const updatedStages = pipelineData.stages.map((stage, i) =>
       i === index ? newStageData : stage,
     )
     setPipelineData((prev) => ({ ...prev, stages: updatedStages }))
   }
 
-  const validateStage = (stage: DraftStageType) => {
+  const validateStage = (stage: PipelineStageRequest) => {
     if (!stage.actions.length) {
       return false
     }
 
+    // action in creation state
+    if (
+      stage.actions.some(
+        (action) =>
+          action.action_type === NEW_PIPELINE_STAGE.actions[0].action_type,
+      )
+    ) {
+      return false
+    }
+
     const segment = stage.actions.find(
```

Comment:
This check does not work anymore if there are several segments and the first one is filled. A `.some` as above should do the trick